libraries/AP_Camera/examples/tracking.py

```diff
@@ -0,0 +1,131 @@
+"""
+Required Installations:
+pip install opencv-python-headless pymavlink ultralytics
+sudo apt-get install libgstreamer1.0-0 gstreamer1.0-tools gstreamer1.0-plugins-base gstreamer1.0-plugins-good
+"""
+
+import cv2
+import threading
+from pymavlink import mavutil
+import math 
+from ultralytics import YOLO
+import argparse
+
+class DroneGimbalControl:
+    def __init__(self, video_source=None):
+        # Set default video source if not provided
+        if video_source is None:
+            video_source = 'udpsrc port=5600 ! application/x-rtp,encoding-name=H264,payload=96 ! rtph264depay ! h264parse ! queue ! avdec_h264 ! videoconvert ! appsink'
+
+        self.master = mavutil.mavlink_connection('127.0.0.1:14560')
+        self.master.wait_heartbeat()
+        print("Heartbeat from the system (system %u component %u)" % 
+             (self.master.target_system, self.master.target_component))
+
+        # Store relevant parameters as object state
+        self.target_system = self.master.target_system
+        self.target_component = self.master.target_component
+        self.flags = 0
+        self.gimbal_device_id = 0
+
+        self.cap = cv2.VideoCapture(video_source, cv2.CAP_GSTREAMER)
+        self.model = YOLO("yolov8n.pt")
+        self.frame_lock = threading.Lock()
+        self.latest_frame = None
+        self.center_y = 0
```

Comment:
Need a lock for these as they are shared state.

Consider a little "Centre" object to contain x, y, and the lock